runtime/envelope/envelope.go

```diff
@@ -202,25 +142,18 @@ func (e *Envelope) toggleProfiling(expected bool) bool {
 	return true
 }
 
-// Run runs the application, restarting it when necessary.
-func (e *Envelope) Run(ctx context.Context) error {
-	if e.opts.GetEnvelopeConn != nil {
-		// The weavelet already started. Set the connection between the
-		// weavelet and the envelope.
-		conn := e.opts.GetEnvelopeConn()
-		e.setConn(conn)
-		return ctx.Err()
-	}
-
-	for r := retry.BeginWithOptions(e.opts.Retry); r.Continue(ctx); {
+// Run runs the application, restarting it according to the policy specified by
+// opts.
+func (e *Envelope) Run(ctx context.Context, opts Options) error {
```

Comment:
The change to move `Options` from a field to an argument of this method isn't strictly necessary. It seemed nice because options were only used by `Run`, and `weavertest` no longer needs to call `Run`, so it is able to exclude options.

If `Options` are likely to be used by other `Envelope` methods, I can restore it as a field.

Reply:
SGTM! :+1: